### ISSUE TITLE
`postfix set-permissions` fails when the overlay Docker Storage Driver is used

### ISSUE THREAD

**Issue by spantaleev** [panubo/docker-postfix#10]
Postfix initialization (`/etc/s6/postfix/run`) includes a call to `postfix set-permissions`.

This will check the `/etc/mailname` file and will cause an error (and ultimately a failure to start this Docker image) if the file has hard links.

One can figure out if the problem will manifest itself by running this command:

```bash
/usr/bin/docker run -it --rm \
--entrypoint=/bin/bash \
panubo/postfix:latest \
-c '/usr/bin/stat /etc/mailname'
```

Why would the file have a different number of hard links? So far, my guess is that it's due to the way the `overlay` Storage Driver for Docker is implemented.

I appears that I can reproduce this problem (the file has 7 hard links for me) when running Docker with the `overlay` Storage Driver.

I cannot reproduce it on machines that use the the `overlay2` or `devicemapper` drivers.

Looking at the [Overlay](https://docs.docker.com/storage/storagedriver/overlayfs-driver/#how-the-overlay2-driver-works) documentation confirms it:

> While the overlay driver only works with a single lower OverlayFS layer and hence requires hard links for implementation of multi-layered images, the overlay2 driver natively supports up to 128 lower OverlayFS layers.

And here's [another comment](https://github.com/coreos/bugs/issues/285#issuecomment-112145017) that confirms the Postfix problem.

-----

While this is more of a problem caused by:
- a limitation of the `overlay` Docker storage driver
- postfix being overly-precautions about hard links

.. it still makes this image fail on Docker systems using the `overlay` driver.

If you'd like to support those, this image may employ some workaround.

It seems like the following code (to be executed before `postfix set-permissions`) should be able to work-around the problem, by making sure that a new file is created:

```bash
echo 'mail.example.com' > /etc/mailname
```

**Comment by spantaleev** [comment]
I just noticed that `/etc/s6/postfix/run` already updates `/etc/mailname` like this:

```bash
echo "${MAILNAME}" > /etc/mailname
```

This is equivalent to the workaround proposed above.
Unfortunately, it's done a little too late (after `postfix set-permissions`).
If it were moved before it (or rather `postfix set-permissions` were moved toward the bottom of the file), this would get solved.

-------

On a related note, instead of writing to the file (`echo "${MAILNAME}" > /etc/mailname`), the same thing can be achieved using:

```bash
postconf -e myorigin="${MAILNAME}"
```

It's probably better to change to that, as it makes the `/etc/s6/postfix/run` file more consistent (everything being done with `postconf`).
